File: test-platform-v2/frontend/src/api/reportTemplate.ts
import api from './client'

export interface TemplateSection {
  key: string
  label: string
  enabled: boolean
  order: number
}

export interface ReportTemplate {
  id: number
  project_id: number
  name: string
  description: string
  sections: TemplateSection[]
  is_default: boolean
  created_at: string | null
  updated_at: string | null
}

export async function fetchTemplates(): Promise<{ total: number; items: ReportTemplate[] }> {
  return api.get('/report-templates') as unknown as Promise<{ total: number; items: ReportTemplate[] }>
}

export async function createTemplate(body: {
  name: string
  description?: string
  sections?: TemplateSection[]
  is_default?: boolean
}): Promise<ReportTemplate> {
  return api.post('/report-templates', body) as unknown as Promise<ReportTemplate>
}

export async function updateTemplate(id: number, body: {
  name?: string
  description?: string
  sections?: TemplateSection[]
  is_default?: boolean
}): Promise<ReportTemplate> {
  return api.put(`/report-templates/${id}`, body) as unknown as Promise<ReportTemplate>
}

export async function deleteTemplate(id: number): Promise<void> {
  return api.delete(`/report-templates/${id}`)
}

File: test-platform-v2/backend/app/schemas/report_template.py
"""ReportTemplate Pydantic schemas — R4a report template feature."""
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class SectionDef(BaseModel):
    """A single section definition within a report template."""
    key: str = ""
    label: str = ""
    enabled: bool = True
    order: int = 0


class TemplateCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: str = ""
    sections: list[SectionDef] = Field(default_factory=list)
    is_default: bool = False


class TemplateUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=100)
    description: str | None = None
    sections: list[SectionDef] | None = None
    is_default: bool | None = None


class TemplateOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int = 0
    name: str = ""
    description: str = ""
    sections: list[SectionDef] = Field(default_factory=list)
    is_default: bool = False
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
